java/arcs/core/policy/Policy.kt

```diff
@@ -71,15 +71,24 @@ data class PolicyTarget(
 
 /** Allowed usages for fields in a schema, see [PolicyFieldProto]. */
 data class PolicyField(
-    // TODO(b/157605232): Resolve the field name to a type.
-    val fieldName: String,
+    /** List of field names leading from the [PolicyTarget] to this nested field. */
+    val fieldPath: List<String>,
     /** Valid usages of this field without redaction. */
     val rawUsages: Set<UsageType> = emptySet(),
     /** Valid usages of this field with redaction first. Maps from redaction label to usages. */
     val redactedUsages: Map<String, Set<UsageType>> = emptyMap(),
     val subfields: List<PolicyField> = emptyList(),
     val annotations: List<Annotation> = emptyList()
-)
+) {
+    init {
+        subfields.forEach { subfield ->
```

Comment:
Instead of having a List<FieldName>, would it be better to keep a `parentField` property and reconstruct the `FieldPath` as needed?  The construction gets tricky as you cannot set the `parentField` property until the parent is constructed. You will need a builder pattern to get around this.

Reply:
That makes it really tricky to create PolicyFields in tests. If the parent holds a reference to the child (subfields) and the child holds a reference to the parent (parentField) then you can't actually create either object fully until the other one has been created. It doesn't work nicely with immutable classes at all... 

I think what we have here is better tbh.